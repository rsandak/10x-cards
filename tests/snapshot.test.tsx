import { describe, it, expect } from "vitest";
import { render } from "@testing-library/react";
import React from "react";

// Przykładowy komponent do testów snapshot
interface CardProps {
  title: string;
  content: string;
  footer?: string;
}

function Card({ title, content, footer }: CardProps) {
  return (
    <div className="card p-4 bg-white shadow-md rounded-md">
      <h3 className="text-lg font-semibold">{title}</h3>
      <p className="mt-2 text-gray-600">{content}</p>
      {footer && <div className="mt-4 pt-2 border-t text-sm text-gray-500">{footer}</div>}
    </div>
  );
}

describe("Card component snapshots", () => {
  it("renders basic card correctly", () => {
    // Arrange
    const { container } = render(<Card title="Test Card" content="This is test content for snapshot" />);

    // Assert - porównanie z zapisanym snapshotem
    expect(container).toMatchSnapshot();
  });

  it("renders card with footer correctly", () => {
    // Arrange
    const { container } = render(
      <Card title="Card with Footer" content="This card has a footer section" footer="Footer text goes here" />
    );

    // Assert - porównanie z zapisanym snapshotem
    expect(container).toMatchSnapshot();
  });
});
