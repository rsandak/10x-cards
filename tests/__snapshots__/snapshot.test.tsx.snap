// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Card component snapshots > renders basic card correctly 1`] = `
<div>
  <div
    class="card p-4 bg-white shadow-md rounded-md"
  >
    <h3
      class="text-lg font-semibold"
    >
      Test Card
    </h3>
    <p
      class="mt-2 text-gray-600"
    >
      This is test content for snapshot
    </p>
  </div>
</div>
`;

exports[`Card component snapshots > renders card with footer correctly 1`] = `
<div>
  <div
    class="card p-4 bg-white shadow-md rounded-md"
  >
    <h3
      class="text-lg font-semibold"
    >
      Card with Footer
    </h3>
    <p
      class="mt-2 text-gray-600"
    >
      This card has a footer section
    </p>
    <div
      class="mt-4 pt-2 border-t text-sm text-gray-500"
    >
      Footer text goes here
    </div>
  </div>
</div>
`;
